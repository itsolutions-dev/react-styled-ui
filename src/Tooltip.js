import styled, { css } from './styledComponent';

const Tooltip = styled({
  component: 'div',
  style: css`
    display: inline-block;
    position: relative;
    color: inherit !important;
    background: transparent !important;

    &::before {
      display: none;
      content: "${props => props.text !== undefined ? String(props.text).replace(/\n/g, '\\A') : ''}";
      position: absolute;
      bottom: 100%;
      left: 50%;
      transform: translate(-50%, -4px);
      white-space: pre;
      font-size: 12px;
      padding-top: 4px;
      padding-bottom: 4px;
      padding-left: 8px;
      padding-right: 8px;
      color: ${props => props.color !== undefined ? props.color : '#fff'};
      background: ${props => props.background !== undefined ? props.background : '#000'};
      border-radius: ${props => props.theme.radius};
    }

    &::after {
      display: none;
      position: absolute;
      bottom: 100%;
      left: 50%;
      transform: translate(-50%, 8px);
      content: " ";
      border-width: 6px;
      border-style: solid;
      border-color: transparent;
      border-top-color: ${props => props.background !== undefined ? props.background : '#000'};
    }

    &:hover {
      &::before, &::after {
        display: block;
      }
    }
  `,
});

export default Tooltip;
