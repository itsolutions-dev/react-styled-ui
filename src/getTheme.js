import defaultTheme from './theme';
import deepMerge from './utils/deepMerge';

const getTheme = deepMerge.bind(null, defaultTheme);

export default getTheme;
