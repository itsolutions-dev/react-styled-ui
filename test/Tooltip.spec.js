import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import getTheme from '../src/getTheme';
import Tooltip from '../src/Tooltip';

const theme = getTheme({
  radius: '7px',
});

describe('Tooltip', () => {
  test('should render a Tooltip', () => {
    const component = ReactTestRenderer.create(<Tooltip theme={theme} />);
    expect(component).toBeDefined();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should have a div tag', () => {
    const component = ReactTestRenderer.create(<Tooltip theme={theme} />).toJSON();
    expect(component.type).toEqual('div');
  });

  test('should have a border-radius', () => {
    const component = ReactTestRenderer.create(
      <Tooltip
        theme={theme}
      />,
    );
    expect({
      component,
      modifier: '&::before',
    }).toHaveStyleRule('border-radius', '7px');
  });

  test('should have the provided text', () => {
    const component = ReactTestRenderer.create(
      <Tooltip
        theme={theme}
        text="foo"
      />,
    );
    expect({
      component,
      modifier: '&::before',
    }).toHaveStyleRule('content', '"foo"');
  });

  test('should handle newlines in text', () => {
    const component = ReactTestRenderer.create(
      <Tooltip
        theme={theme}
        text={
          `Hello
          World!`}
      />,
    );
    expect({
      component,
      modifier: '&::before',
    }).toHaveStyleRule('content', '"Hello\\A          World!"');
  });

  test('should have the provided background', () => {
    const component = ReactTestRenderer.create(
      <Tooltip
        theme={theme}
        background="white"
      />,
    );
    expect({
      component,
      modifier: '&::before',
    }).toHaveStyleRule('background', 'white');
  });

  test('should have the provided color', () => {
    const component = ReactTestRenderer.create(
      <Tooltip
        theme={theme}
        color="white"
      />,
    );
    expect({
      component,
      modifier: '&::before',
    }).toHaveStyleRule('color', 'white');
  });

  test('should have a default text', () => {
    const component = ReactTestRenderer.create(
      <Tooltip
        theme={theme}
      />,
    );
    expect({
      component,
      modifier: '&::before',
    }).toHaveStyleRule('content', '""');
  });

  test('should have a default background', () => {
    const component = ReactTestRenderer.create(
      <Tooltip
        theme={theme}
      />,
    );
    expect({
      component,
      modifier: '&::before',
    }).toHaveStyleRule('background', '#000');
  });

  test('should have a default color', () => {
    const component = ReactTestRenderer.create(
      <Tooltip
        theme={theme}
      />,
    );
    expect({
      component,
      modifier: '&::before',
    }).toHaveStyleRule('color', '#fff');
  });
});
