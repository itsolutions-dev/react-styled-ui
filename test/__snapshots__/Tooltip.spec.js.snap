// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should render a Tooltip 1`] = `
.c0 {
  display: inline-block;
  position: relative;
  color: inherit !important;
  background: transparent !important;
}

.c0::before {
  display: none;
  content: "";
  position: absolute;
  bottom: 100%;
  left: 50%;
  -webkit-transform: translate(-50%,-4px);
  -ms-transform: translate(-50%,-4px);
  transform: translate(-50%,-4px);
  white-space: pre;
  font-size: 12px;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  color: #fff;
  background: #000;
  border-radius: 7px;
}

.c0::after {
  display: none;
  position: absolute;
  bottom: 100%;
  left: 50%;
  -webkit-transform: translate(-50%,8px);
  -ms-transform: translate(-50%,8px);
  transform: translate(-50%,8px);
  content: " ";
  border-width: 6px;
  border-style: solid;
  border-color: transparent;
  border-top-color: #000;
}

.c0:hover::before,
.c0:hover::after {
  display: block;
}

<div
  className="c0"
/>
`;
